### ISSUE TITLE
Default arguments can be used to bypass the termination checker

### ISSUE THREAD

**Issue by paulcadman** [anoma/juvix#2460]
The following program compiles without error, but loops at runtime:

```
module M;

type T := t;

g : T := f;
f {a : T := g} : T := a;

main : T := f;
```

NB: The following loops the typechecker (which is OK?):

```
module M;

type T := t;

f {a : T := f} : T := a;
```

**Comment by janmasrovira** [comment]
This will be solved by #2446. I've added this test there

**Comment by paulcadman** [comment]
Closed by https://github.com/anoma/juvix/pull/2446
